packages/turf-boolean-is-valid/index.ts

```diff
@@ -0,0 +1,99 @@
+import { segmentEach } from '@turf/helpers';
+import { getGeom, getCoords, getType } from '@turf/invariant';
+import { polygon, lineString } from '@turf/helpers';
+import booleanDisjoint from '@turf/boolean-disjoint';
+import booleanCrosses from '@turf/boolean-crosses';
+import lineIntersect from '@turf/line-intersect';
+import isPointOnLine from '@turf/boolean-point-on-line';
+
+/**
+ * booleanIsValid checks if the geometry is a valid according to the OGC Simple Feature Specification.
+ *
+ * @name booleanIsValid
+ * @param {Geometry|Feature<any>} feature GeoJSON Feature or Geometry
+ * @returns {boolean} true/false
+ * @example
+ * var line = turf.lineString([[1, 1], [1, 2], [1, 3], [1, 4]]);
+ *
+ * turf.booleanIsValid(line, point);
+ * //=true
+ */
+export default function booleanIsValid(feature: Feature<any> | Geometry) {
+    const geom = getGeom(feature);
+    const type = getType(feature);
+    const coords = getCoords(feature);
+
+    switch (type) {
+    case 'Point':
+        return coords.length > 1;
+    case 'MultiPoint':
+        for (var i = 0; i < coords.length; i++) {
+            if (coords[i].length < 2) return false;
```

Comment:
These switch cases get tricky... 🤔 Hard to properly handle these scenarios in a "strongly typed" way.

Easy solution would be to convert `coords` to `any`:

```
const coords: any = getCoords(feature);
```